File: src/main/java/ValidadorContrasenias/ValidacionCompararContraPeores.java
package ValidadorContrasenias;

import java.io.File;
import java.io.FileNotFoundException;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.Scanner;

public class ValidacionCompararContraPeores implements Validable{
    public ValidacionCompararContraPeores() {
    }

    @Override
    public boolean esValida(String clave) {
        Path path = Paths.get("src/main/java/ValidadorContrasenias/10000-most-common-passwords.txt");

        File archivo = new File(path.toAbsolutePath().toString());
        Scanner lector = null;
        try {
            lector = new Scanner(archivo);
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }

        while(lector.hasNextLine()){

            if(lector.nextLine().equals(clave)){
                return false;
            }
        }
        return true;
    }
}

File: src/main/ts/ar/edu/utn/frba/dds/domain/ValidadorContraseñas/ValidacionCompararContraPeores.ts
import * as fs from 'fs';
import { Validable } from './Validable';

class ValidacionCompararContraPeores implements Validable {
    #contraseñasComunes: Array<string>;

    constructor() {
        this.#contraseñasComunes = new Array<string>();
    }

    esValida(clave: string): boolean {
        if (this.#contraseñasComunes == null) {
            this.#leerContraseñasDesdeArchivo();
        }
        return this.#contraseñasComunes.includes(clave);
    };

    #leerContraseñasDesdeArchivo(): void {
        const file = fs.readFileSync("./10000-most-common-passwords.txt", "utf-8");
        this.#contraseñasComunes = file.split('\r\n');
    };
}